### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by mpalyga** [Crudelis98/Klarta-Forste-3-WiFi-Mod#1]
Hej 
  module works fantasticly well but few questions from my side 
   - is project abandoned as couple of actions still pending (eg. full HA discovery, etc)  ?
   -  is a plan to to upgrade tasmota to higner version ? 
   - is a chance to add full version of code to git repo (i have few modifictaion now) ? 

regards m.

**Comment by Crudelis98** [comment]
is project abandoned as couple of actions still pending (eg. full HA discovery, etc) ?
I'm in Neaples, doing my master degree. I will push some updates during Christmas.
is a plan to to upgrade tasmota to higner version ?
Yes, I will push some updates during Christmas.
is a chance to add full version of code to git repo (i have few modifictaion now) ?
It will be done, I need to extract the core. There was a lot of debbuging functions that are not in use but they were needed for me to understand how Klatra was made.

Also there is one bug that could make module to reboot after 20 sec. It is fixed. I will try push update this week.

**Comment by shalak** [comment]
I'm also interested in this project. Do you know if it will support Klarta Forste 4?

**Comment by Crudelis98** [comment]
I don't have KF 4 and it supports Wi-Fi, doesnt it?

**Comment by shalak** [comment]
> I don't have KF 4 and it supports Wi-Fi, doesnt it?

Yeah, but only via cloud. That's a no-go for me :) (and I couldn't find any reliable information on making KF4 work with Local Tuya, which is almost cloud-free)

**Comment by Crudelis98** [comment]
I expected you would say that, thats why I went for Klarta 3, very good quality for the price. Since I would need to re-flash Wi-Fi eneabled purifier to get local access and disable cloud, I decided to mod device without Wi-Fi.

**Comment by shalak** [comment]
Unfortunately, KF3 is hard to come by, also I believe that KF4 has other improvements, aside from WiFi.

Anyway, I'm looking forward for updates to this repo.

**Comment by Crudelis98** [comment]
If someone have KF4, please tell me what mcu is inside. I'm abroad so its impossible for me to get one.

**Comment by mpalyga** [comment]
Hej
  i know ... still Christmas time almost .. but maybe any news for deployment/enhancement ? 
  btw. KF4 is Tuya based so easily to make it integrated with HA .. not sure about flashing to Tasmota (i love it) - KF3 works well - only some pain is pending full HA integration :)

/ regards m.

**Comment by Crudelis98** [comment]
Full HA integration depends on Tasmota to HA integration addon. I didn't forget, I'm coding my own webpage for this product and maybe others in the future including OTA updates and discord server.

**Comment by mpalyga** [comment]
hej
 an update ? not sure but chat gpt can code own webpage very quickly ... is a chance to publish the code ? i can help to integrate with new version of tasmota and complete HA full integration .. 

/regards m.

**Comment by Crudelis98** [comment]
Hello,
As I said, Im not home unfortunatelly. Can not edit or upload the code, I know it has been a year... I didnt expect the master thesis to be so demanding here. Anyway I'm finishing my education, so please just a little bit longer, meanwhile I have seen push buttons in HA from 2022. They didnt exist back when I was selling the boards. Im not sure if they support auto discovery. So it possible that I still can not implement it, but the device supports api for this allready. So while you are waiting just create few push buttons and assign klarta mqtt api functions.
